shell/server/admin-server.js

```diff
@@ -473,6 +473,90 @@ Meteor.publish("allUsers", function (token) {
   return Meteor.users.find();
 });
 
+Meteor.publish("adminUserDetails", function (userId) {
+  if (!authorizedAsAdmin(undefined, this.userId)) return [];
+
+  // Reactive publish of any identities owned by
+  const identitySubs = {};
+  const accountId = userId;
+
+  const unrefIdentity = (identityId) => {
+    const observeHandle = identitySubs[identityId];
+    identitySubs[identityId] = undefined;
+    observeHandle.stop();
+    this.removed("users", identityId);
+  };
+
+  const refIdentity = (identityId) => {
+    const cursor = Meteor.users.find({ _id: identityId });
+    const observeHandle = cursor.observe({
+      added: (doc) => {
+        this.added("users", doc._id, doc);
+      },
+
+      changed: (doc) => {
+        this.changed("users", newDoc._id, newDoc);
+      },
+
+      removed: (doc) => {
+        this.removed("users", oldDoc._id);
+      },
+    });
+
+    identitySubs[identityId] = observeHandle;
```

Comment:
OK, I think you should add some defensive code to detect a double-subscribe (and double-unsubscribe). If somehow someone ends up with the same identity listed twice on their account, it shouldn't cause the server to leak mongo subscriptions. Plus I could easily imagine this code changing later on to be used in a different way, and the name `refIdentity` sort of implies refcounting, so I'd hate for someone to assume that's what it does and have a silent leak...